### DISCUSSION TITLE
refresh token

### DISCUSSION THREAD

**Discussion by massimopalmieri** [sergiodxa/remix-auth#102] [Q&A] [answered]
apologies for the silly question, but I haven't managed to figure it out yet. How do I get a fresh token (google strategy) once it expires?

**Comment by sergiodxa** [comment, marked as answer]
In the verify callback you pass to the GoogleStrategy when you instantiate you receive an access token, then a refresh token, then extra params from Google and finally the user profile.

You can keep both the access token and refresh token and once you need to use the access token and it fails because it expired you can use the refresh token to get a new one and keep the new one.

Doing that it's outside the scope of the package, you need to do that in your loaders or actions.

**Reply by massimopalmieri** [reply]
thanks @sergiodxa! I was able to access the refresh token in the verify callback, but for some reason, I thought that there was a way to  get a refreshed token with the library rather than manually. Thanks for clarifying.
